docs/RFCS/20220628_invisible_index.md

```diff
@@ -0,0 +1,468 @@
+no risks. Firstly, just because an index is not used during this observation
+period, this does not mean it will not be used in the future. Secondly,
+invisible indexes are still used behind the scene by the optimizer for any
+constraint check and maintenance purposes (more details below). In that case,
+you cannot expect the database to behave in the exactly same way as dropping an
+index.
+
+### 4. Make indexes only available to specific queries.
+If you want to make an index available only for specific queries, if only a
+certain part of your application requires an index, or if you know a single
+query is causing problems and creating an index would help, you can use an
+invisible index and make this index available only to queries you want. This
+leaves the rest of your application unaffected.
+
+### 5. Index foreign key columns on a child table.
+As mentioned earlier, invisible indexes are actually used for any constraint
+check or maintenance purposes. This unlocks another use case. Invisible indexes
+on foreign key columns can still be used to prevent full table locking and
+performance issues.
+
+When a parent table performs update or delete operations, the optimizer must
+perform a FK check on the child table to ensure that there are no rows in the
+child table referencing the columns being deleted or updated. If there are such
+rows, this operation will fail. If the foreig key column on the child table is
+not indexed, this check would require a full table level lock. This means that
+if there is another operation on any rows in the child table, the operation will
+still hang. On occasions, this could even lead to deadlocks.
+
+However, having an index on foreign key columns, regardless of its visibility,
+will only lead to a row level lock. This means that the operation will no longer
+hang if there is another operation on an unrelated row in the child table. But
+it will still hang if the operation is performed on the same row. This is one of
+the cases where people should be cautious. Dropping such invisible indexes will
+lead to a full table lock which is very different from marking an index
+invisible.
+
+
+
+# Implementation Decisions
+Please let me know if you have any questions or different opinions on these
+decisions. It should be relatively easy to make a change. Any feedback would be
+greatly appreciated : )
+
+### Conclusion:
+- Users can create an invisible index or alter an existing index to be invisible.
+- Indexes are visible by default.
+- Primary indexes are not allowed to be invisible.
+- Constraints cannot be created with invisible indexes. Creating a unique or a foreign key constraint with invisible indexes is not supported.
+- Partial invisible indexes or inverted invisible indexes are both supported. The behavior is as expected.
+- Queries can be instructed to use invisible indexes explicitly through force index or index hinting.
+- Recreating an invisible index will make the index visible by default.
```

Comment:
Do you mean `DROP INDEX ...; CREATE INDEX ...;`? I think this is obvious enough to omit here. Plus, it's a little confusing because if I recreate the index identically with the `NOT VISIBLE` syntax, it would be invisible.